R/importNet.Multinet.R

```diff
@@ -0,0 +1,145 @@
+# importNet.MULTINET.R
+
+#' Import network data from BIOGRID, ENCODE data, KEGG, and SignaLink database 
+#' (Multinet interactions).
+#'
+#' \code{importNet.MULTINET} imports network edges from a MULTINET database file,
+#' 
+#' @section Networks:
+#'   MULTINET "protein.links.detailed.v10.txt" files contain
+#'   several protein networks: neighborhood, fusion, cooccurence, coexpression,
+#'   experimental, database, textmining, and combined_score. However this
+#'   function is not restricted to these types, but will read one network for
+#'   which the column name is requested in the function's net parameter. This
+#'   allows users to define their own column.
+#' @section The gG object:
+#'   The function returns a rete gG object, a weighted,
+#'   directed, simple igraph graph in which HGNC gene symbols are vertex names,
+#'   and the edge attributes $weight hold the network scores. Graph attributes
+#'   hold metadata; use igraph::graph_attr(gG) to return: $version: the gG
+#'   object version; $logFile: the filename to which log information was
+#'   written; $inFile: the input filename of MULTINET data; $call: the complete
+#'   function call with expanded arguments; and $date: when the gG object was
+#'   created.
+#'
+#' @param fName Filename of a MultiNet Multinet.interactions.network_presence.txt file.
+#' @param net The requested network. This must be a MULTINET that exists in the
+#'   header. The default is "PPI".
+#' @param verbose Controls whether to print summary information after import. 
+#'   Defaults to TRUE.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return a weighted, directed, simple igraph graph which is a rete gG object.
+#'
+#' @family ImportNet.STRING, importNet.MultiNet, importNet.MITAB
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to map ENSP
+#'   IDs to gene symbols.
+#'
+#'   ## @examples ## \dontrun{ ## importNet.MULTINET(IN, OUT) ## }
+#' @export
+importNet.MultiNet <- function(fName,
+                             net = "PPI",
+                             verbose = TRUE,
+                             silent = FALSE,
+                             writeLog = TRUE) {
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fName,        like = "FILE_E",   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(net,          like = "a",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(verbose,      like = logical(1), checkSize = TRUE))
```

Comment:
Hey Boris, I made the same comment earlier. You might need to refresh to see them.